### ISSUE TITLE
wrong installation path dependencies

### ISSUE THREAD

**Issue by lferraz** [NVIDIA/VideoProcessingFramework#213]
just after `make && make` install if you remove the build folder the lib fails.

After running `ldd` on the `PyNvCodec` lib you can see that the libs `libTC.so` and `libTC_CORE.so` point to the build directory.

e.g.
 installing the lib in `VideoProcessingFramework/install/bin`

`ldd PyNvCodec.cpython-37m-x86_64-linux-gnu.so `

shows:
```
 libTC.so =>MYPATH/VideoProcessingFramework/build/PyNvCodec/TC/libTC.so (0x00007faee7916000)
libTC_CORE.so => MYPATH/VideoProcessingFramework/build/PyNvCodec/TC/TC_CORE/libTC_CORE.so (0x00007faee73a9000) 
```

**Comment by rarzumanyan** [comment]
Hi @lferraz 

What's the output of `echo $LD_LIBRARY_PATH` on your machine?

**Comment by lferraz** [comment]
I tested in 3 places.
- inside docker:
```
root@b0c8cb8a80d8:/src# echo $LD_LIBRARY_PATH
/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/cuda/compat/
```
I added the last path to be able to get acces to `libcuda.so.1`

- In other 2 machines:
```

echo $LD_LIBRARY_PATH
(empty)
``` 

in all the situations the path of the dependencies returned by `ldd` is wrong. It always point to the `build` folder.

**Comment by rarzumanyan** [comment]
Thanks for the update @lferraz.

CMake `INSTALL` target ends up copying files to the `CMAKE_INSTALL_PREFIX` directory, it has nothing to do with Linux dynamic linker:
https://github.com/NVIDIA/VideoProcessingFramework/blob/906b6dc43e6be99284c24e382cf5fc93196d99c7/CMakeLists.txt#L48-L49

You can find more information on dynamic libraries search path by calling `man ldd` and `man ld`.
Also take a look at [Stack Exchange page](https://unix.stackexchange.com/questions/367600/what-is-the-order-that-linuxs-dynamic-linker-searches-paths-in).

Usually this kind of a problem can be solved by exporting `LD_LIBRARY_PATH` before running the script:
```bash
cd ~/VideoProcessingFramework/install/bin
export LD_LIBRARY_PATH=$(pwd):$LD_LIBRARY_PATH
python SampleDecode.py 0 input_video.avi output.nv12
```

Alternatively you can export the `LD_LIBRARY_PATH` env. variable at `~/.bashrc` file.

**Comment by lferraz** [comment]
Hi @rarzumanyan , 

I feel that's not the problem. In my opinion the problem is that PyNvCodec lib always looks for (even after the installation) the TC lib in `build/PyNvCodec/TC/libTC.so` and TC_CORE lib in `build/PyNvCodec/TC/TC_CORE/libTC_CORE.so` instead of `install/bin/libTC.so` and  `install/bin/libTC_CORE.so`.

I did a little bit of research and I think the problem is in the `CMakeList.txt`. 
Basically I read that the paths should be somehow relative. 
https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-relocatable-packages
I tried to understand how CMAKE works but it is not trivial at all. :S

**Comment by rarzumanyan** [comment]
Thanks for the update @lferraz.

I'm running out of depth here TBH so I'm gonna need some time to go through the CMake packaging docs.
If you want to experiment with it, I think you need to take a look here:
https://github.com/NVIDIA/VideoProcessingFramework/blob/ab6254236bb699ea2b2d5009d06b0ee137298ce3/PyNvCodec/CMakeLists.txt#L58

**Comment by lferraz** [comment]
HI @rarzumanyan ,

it is not a high priority, In my case I can deal with it. But it is something that it is unexpected.

I was like a couple of hours looking at the cmake files and the documentation and I couldn't get any clear conclusion. I think we need an expert using CMake XD

Anyway thank you so much for you effort and time!!!!

**Comment by contentis** [comment]
@lferraz Can this be closed?

**Comment by lferraz** [comment]
in fact this is not addressed but using some tricks I avoid the issue. So, feel free to close it @contentis
